In [1]:
#importing libraries

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
#importing data

df = pd.read_csv('time_series_covid_19_confirmed.csv')
df.head()

Province/State Country/Region      Lat     Long  1/22/20  1/23/20  1/24/20  \
0            NaN    Afghanistan  33.0000  65.0000        0        0        0   
1            NaN        Albania  41.1533  20.1683        0        0        0   
2            NaN        Algeria  28.0339   1.6596        0        0        0   
3            NaN        Andorra  42.5063   1.5218        0        0        0   
4            NaN         Angola -11.2027  17.8739        0        0        0   

   1/25/20  1/26/20  1/27/20   ...    3/23/20  3/24/20  3/25/20  3/26/20  \
0        0        0        0   ...         40       74       84       94   
1        0        0        0   ...        104      123      146      174   
2        0        0        0   ...        230      264      302      367   
3        0        0        0   ...        133      164      188      224   
4        0        0        0   ...          3        3        3        4   

   3/27/20  3/28/20  3/29/20  3/30/20  3/31/20  4/1/20  
0      110      110      120      170      174     237  
1      186      197      212      223      243     259  
2      409      454      511      584      716     847  
3      267      308      334      370      376     390  
4        4        5        7        7        7       8  

[5 rows x 75 columns]

In [3]:
df.describe()

Lat        Long     1/22/20     1/23/20     1/24/20     1/25/20  \
count  256.000000  256.000000  256.000000  256.000000  256.000000  256.000000   
mean    21.823943   23.158832    2.167969    2.554688    3.675781    5.601562   
std     24.484347   71.120822   27.805753   27.905108   34.737565   48.343840   
min    -41.454500 -135.000000    0.000000    0.000000    0.000000    0.000000   
25%      7.844875  -16.237775    0.000000    0.000000    0.000000    0.000000   
50%     23.692500   20.921188    0.000000    0.000000    0.000000    0.000000   
75%     41.227200   84.497525    0.000000    0.000000    0.000000    0.000000   
max     71.706900  178.065000  444.000000  444.000000  549.000000  761.000000   

           1/26/20      1/27/20      1/28/20      1/29/20      ...        \
count   256.000000   256.000000   256.000000   256.000000      ...         
mean      8.273438    11.433594    21.789062    24.085938      ...         
std      67.557350    91.021068   223.404115   224.758713      ...         
min       0.000000     0.000000     0.000000     0.000000      ...         
25%       0.000000     0.000000     0.000000     0.000000      ...         
50%       0.000000     0.000000     0.000000     0.000000      ...         
75%       0.000000     0.000000     0.000000     0.000000      ...         
max    1058.000000  1423.000000  3554.000000  3554.000000      ...         

            3/23/20       3/24/20       3/25/20       3/26/20        3/27/20  \
count    256.000000    256.000000    256.000000    256.000000     256.000000   
mean    1477.480469   1632.988281   1826.769531   2068.714844    2317.542969   
std     7239.371162   7863.701629   8698.423123   9828.145417   11007.250578   
min        0.000000      0.000000      0.000000      0.000000       0.000000   
25%        5.000000      6.000000      8.000000     11.000000      11.000000   
50%       74.000000     82.500000     92.000000    100.500000     106.500000   
75%      358.500000    398.250000    446.250000    496.750000     570.250000   
max    67800.000000  69176.000000  74386.000000  83836.000000  101657.000000   

             3/28/20        3/29/20        3/30/20        3/31/20  \
count     256.000000     256.000000     256.000000     256.000000   
mean     2580.882812    2812.957031    3056.113281    3349.558594   
std     12305.356049   13521.252906   14849.525365   16487.495451   
min         0.000000       0.000000       0.000000       0.000000   
25%        12.000000      15.000000      16.500000      17.750000   
50%       119.500000     137.000000     147.000000     152.000000   
75%       629.000000     661.750000     697.750000     743.500000   
max    121478.000000  140886.000000  161807.000000  188172.000000   

              4/1/20  
count     256.000000  
mean     3642.988281  
std     18126.408076  
min         0.000000  
25%        19.000000  
50%       173.000000  
75%       799.000000  
max    213372.000000  

[8 rows x 73 columns]

In [4]:
df['Province/State'].isnull().sum()

177

# Creating a bar chart race animation

With 177 missing rows it's not a good idea to keep the Province/State column in the data.

Also latitude and longitude will not be necessary for creating a bar chart race animation.

In [5]:
chart_df = df.drop(['Province/State','Lat', 'Long'],axis=1)
chart_df.head()

Country/Region  1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  \
0    Afghanistan        0        0        0        0        0        0   
1        Albania        0        0        0        0        0        0   
2        Algeria        0        0        0        0        0        0   
3        Andorra        0        0        0        0        0        0   
4         Angola        0        0        0        0        0        0   

   1/28/20  1/29/20  1/30/20   ...    3/23/20  3/24/20  3/25/20  3/26/20  \
0        0        0        0   ...         40       74       84       94   
1        0        0        0   ...        104      123      146      174   
2        0        0        0   ...        230      264      302      367   
3        0        0        0   ...        133      164      188      224   
4        0        0        0   ...          3        3        3        4   

   3/27/20  3/28/20  3/29/20  3/30/20  3/31/20  4/1/20  
0      110      110      120      170      174     237  
1      186      197      212      223      243     259  
2      409      454      511      584      716     847  
3      267      308      334      370      376     390  
4        4        5        7        7        7       8  

[5 rows x 72 columns]

In [6]:
chart_df.shape

(256, 72)

Let's rename the Country/Region column as Country for our future use

In [7]:
chart_df.rename(columns = {'Country/Region':'Country'},inplace=True)

In [8]:
chart_df.columns

Index(['Country', '1/22/20', '1/23/20', '1/24/20', '1/25/20', '1/26/20',
       '1/27/20', '1/28/20', '1/29/20', '1/30/20', '1/31/20', '2/1/20',
       '2/2/20', '2/3/20', '2/4/20', '2/5/20', '2/6/20', '2/7/20', '2/8/20',
       '2/9/20', '2/10/20', '2/11/20', '2/12/20', '2/13/20', '2/14/20',
       '2/15/20', '2/16/20', '2/17/20', '2/18/20', '2/19/20', '2/20/20',
       '2/21/20', '2/22/20', '2/23/20', '2/24/20', '2/25/20', '2/26/20',
       '2/27/20', '2/28/20', '2/29/20', '3/1/20', '3/2/20', '3/3/20', '3/4/20',
       '3/5/20', '3/6/20', '3/7/20', '3/8/20', '3/9/20', '3/10/20', '3/11/20',
       '3/12/20', '3/13/20', '3/14/20', '3/15/20', '3/16/20', '3/17/20',
       '3/18/20', '3/19/20', '3/20/20', '3/21/20', '3/22/20', '3/23/20',
       '3/24/20', '3/25/20', '3/26/20', '3/27/20', '3/28/20', '3/29/20',
       '3/30/20', '3/31/20', '4/1/20'],
      dtype='object')

Let's check the number of countries affected by Covid 19

In [9]:
chart_df['Country'].value_counts()

China                     33
Canada                    15
France                    10
United Kingdom            10
Australia                  8
Netherlands                4
Denmark                    3
Italy                      1
Serbia                     1
Benin                      1
San Marino                 1
Switzerland                1
Norway                     1
Cyprus                     1
Rwanda                     1
Czechia                    1
Madagascar                 1
Grenada                    1
Holy See                   1
Papua New Guinea           1
India                      1
El Salvador                1
Djibouti                   1
Cabo Verde                 1
Slovenia                   1
Ghana                      1
Bulgaria                   1
Morocco                    1
Ukraine                    1
Mauritania                 1
                          ..
Liechtenstein              1
Singapore                  1
Slovakia                   1
Honduras      

So the number of affected countries seem to be 180 but there are 256 columns. So it is very correct to assume that some countries are repeated.

Well thinking more it will be clear that the repetition maybe due to the Provine/State values because more than one provine may belong to the same country

Since all the province belong to the same country it will be safe to get the sum of number of confirmed cases for all the repititive values in Country name.

In [10]:
chart_df.head(10)

Country  1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  \
0          Afghanistan        0        0        0        0        0        0   
1              Albania        0        0        0        0        0        0   
2              Algeria        0        0        0        0        0        0   
3              Andorra        0        0        0        0        0        0   
4               Angola        0        0        0        0        0        0   
5  Antigua and Barbuda        0        0        0        0        0        0   
6            Argentina        0        0        0        0        0        0   
7              Armenia        0        0        0        0        0        0   
8            Australia        0        0        0        0        0        0   
9            Australia        0        0        0        0        3        4   

   1/28/20  1/29/20  1/30/20   ...    3/23/20  3/24/20  3/25/20  3/26/20  \
0        0        0        0   ...         40       74       84       94   
1        0        0        0   ...        104      123      146      174   
2        0        0        0   ...        230      264      302      367   
3        0        0        0   ...        133      164      188      224   
4        0        0        0   ...          3        3        3        4   
5        0        0        0   ...          3        3        3        7   
6        0        0        0   ...        301      387      387      502   
7        0        0        0   ...        235      249      265      290   
8        0        0        0   ...         32       39       39       53   
9        4        4        4   ...        669      818     1029     1219   

   3/27/20  3/28/20  3/29/20  3/30/20  3/31/20  4/1/20  
0      110      110      120      170      174     237  
1      186      197      212      223      243     259  
2      409      454      511      584      716     847  
3      267      308      334      370      376     390  
4        4        5        7        7        7       8  
5        7        7        7        7        7       7  
6      589      690      745      820     1054    1054  
7      329      407      424      482      532     571  
8       62       71       77       78       80      84  
9     1405     1617     1791     2032     2032    2182  

[10 rows x 72 columns]

In [11]:
chart_df = chart_df.groupby('Country').agg(np.sum)
chart_df = chart_df.reset_index().rename(columns = {'index':'Country'})
chart_df.head(10)

Country  1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  \
0          Afghanistan        0        0        0        0        0        0   
1              Albania        0        0        0        0        0        0   
2              Algeria        0        0        0        0        0        0   
3              Andorra        0        0        0        0        0        0   
4               Angola        0        0        0        0        0        0   
5  Antigua and Barbuda        0        0        0        0        0        0   
6            Argentina        0        0        0        0        0        0   
7              Armenia        0        0        0        0        0        0   
8            Australia        0        0        0        0        4        5   
9              Austria        0        0        0        0        0        0   

   1/28/20  1/29/20  1/30/20   ...    3/23/20  3/24/20  3/25/20  3/26/20  \
0        0        0        0   ...         40       74       84       94   
1        0        0        0   ...        104      123      146      174   
2        0        0        0   ...        230      264      302      367   
3        0        0        0   ...        133      164      188      224   
4        0        0        0   ...          3        3        3        4   
5        0        0        0   ...          3        3        3        7   
6        0        0        0   ...        301      387      387      502   
7        0        0        0   ...        235      249      265      290   
8        5        6        9   ...       1682     2044     2364     2810   
9        0        0        0   ...       4474     5283     5588     6909   

   3/27/20  3/28/20  3/29/20  3/30/20  3/31/20  4/1/20  
0      110      110      120      170      174     237  
1      186      197      212      223      243     259  
2      409      454      511      584      716     847  
3      267      308      334      370      376     390  
4        4        5        7        7        7       8  
5        7        7        7        7        7       7  
6      589      690      745      820     1054    1054  
7      329      407      424      482      532     571  
8     3143     3640     3984     4361     4559    4862  
9     7657     8271     8788     9618    10180   10711  

[10 rows x 72 columns]

In [12]:
chart_df.shape

(180, 72)

Now the data looks ready for creating a bar chart race

In [13]:
#creating the csv file

chart_df.to_csv('bar_chart_race.csv')